### PR TITLE
smart gate feature implementation

### PR DESCRIPTION
**Pull request by 0xxlegolas**
- Checking the maxDistance before linking 2 gates

### CODE REVIEW

File: mud-contracts/world/src/modules/smart-gate/systems/SmartGate.sol [hunk 1]
```diff
@@ -123,6 +126,10 @@ contract SmartGate is EveSystem, AccessModified {
 
     //TODO: Check if the state is online for both the gates ??
     //TODO: Link the gates only when the distance between 2 gates are less than the max distance
```
Comment (vayan) [inline mud-contracts/world/src/modules/smart-gate/systems/SmartGate.sol]:
```suggestion
```
Done now 😄

File: mud-contracts/world/src/codegen/world/ISmartGate.sol [hunk 2]
```diff
@@ -42,5 +43,5 @@ interface ISmartGate {
 
   function eveworld__isGateLinked(uint256 sourceGateId, uint256 destinationGateId) external view returns (bool);
 
-  function eveworld__calculatDistance(uint256 sourceGateId, uint256 destinationGateId) external view returns (uint256);
+  function eveworld__isWithinRange(uint256 sourceGateId, uint256 destinationGateId) external view returns (bool);
```
Comment (vayan) [inline mud-contracts/world/src/codegen/world/ISmartGate.sol:46]:
```suggestion
  function eveworld__isGateWithinRange(uint256 sourceGateId, uint256 destinationGateId) external view returns (bool);
```

after using `eveworld__inProximity` and `eveworld__agression` I think we should be a bit more verbose

Comment (0xxlegolas) [reply mud-contracts/world/src/codegen/world/ISmartGate.sol:46]: Yes good catch, Will recompile it and push. Thanks

**Review by fc-tezza — APPROVED** [review]
All looks good to me..

My only comment is that to be very safe in the contract logic, in the future we may wan to ensure that the maximum possible distance^2 doesn't not overflow uint256 (and we may want to enforce this on the parameter inputs on the contract level). But other than this, all looking great for the current release.

**Pull request merged**